db/Pos/DB/Rocks/Functions.hs

```diff
@@ -0,0 +1,156 @@
+-- interface.
+
+module Pos.DB.Rocks.Functions
+       (
+       -- * Opening and modifications
+         openRocksDB
+       , closeRocksDB
+       , usingReadOptions
+       , usingWriteOptions
+
+       -- * Reading/writing
+       , rocksGetBytes
+       , rocksPutBytes
+       , rocksDelete
+       , rocksPutBi
+
+       -- * Snapshot
+       , Snapshot (..)
+       , usingSnapshot
+
+       -- * Iteration
+       , rocksIterSource
+       ) where
+
+import           Universum
+
+import           Control.Lens                 (ASetter')
+import           Control.Monad.Trans.Resource (MonadResource)
+import           Data.Conduit                 (ConduitM, Source, bracketP, yield)
+import           Data.Default                 (def)
+import qualified Database.RocksDB             as Rocks
+import           Ether.Internal               (lensOf)
+
+import           Pos.Binary.Class             (Bi, encode)
+import           Pos.DB.Class                 (DBIteratorClass (..), DBTag (..), IterType)
+import           Pos.DB.Functions             (processIterEntry)
+import           Pos.DB.Rocks.Types           (DB (..), MonadRealDB, NodeDBs, getDBByTag)
+
+----------------------------------------------------------------------------
+-- Opening/options
+----------------------------------------------------------------------------
+
+openRocksDB :: MonadIO m => FilePath -> m DB
+openRocksDB fp = DB def def def
+                   <$> Rocks.open fp def
+                        { Rocks.createIfMissing = True
+                        , Rocks.compression     = Rocks.NoCompression
+                        }
+
+
+closeRocksDB :: MonadIO m => DB -> m ()
+closeRocksDB = Rocks.close . rocksDB
+
+usingReadOptions
+    :: MonadRealDB ctx m
+    => Rocks.ReadOptions
+    -> ASetter' NodeDBs DB
+    -> m a
+    -> m a
+usingReadOptions opts l =
+    local (over (lensOf @NodeDBs . l) (\db -> db {rocksReadOpts = opts}))
+
+usingWriteOptions
+    :: MonadRealDB ctx m
+    => Rocks.WriteOptions
+    -> ASetter' NodeDBs DB
+    -> m a
+    -> m a
+usingWriteOptions opts l =
+    local (over (lensOf @NodeDBs . l) (\db -> db {rocksWriteOpts = opts}))
+
+----------------------------------------------------------------------------
+-- Reading/writing
+----------------------------------------------------------------------------
+
+-- | Read ByteString from RocksDb using given key.
+rocksGetBytes :: (MonadIO m) => ByteString -> DB -> m (Maybe ByteString)
+rocksGetBytes key DB {..} = Rocks.get rocksDB rocksReadOpts key
+
+-- | Write ByteString to RocksDB for given key.
+rocksPutBytes :: (MonadIO m) => ByteString -> ByteString -> DB -> m ()
+rocksPutBytes k v DB {..} = Rocks.put rocksDB rocksWriteOpts k v
+
+-- | Delete element from RocksDB for given key.
+rocksDelete :: (MonadIO m) => ByteString -> DB -> m ()
+rocksDelete k DB {..} = Rocks.delete rocksDB rocksWriteOpts k
+
+-- garbage, should be abstracted and hidden
```

Comment:
I will check it. I think i wrote this comment because i didn't figure it out. Maybe i will this time :)